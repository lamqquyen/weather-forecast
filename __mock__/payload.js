export const locationPayload = [
  {
    title: 'San Francisco',
    location_type: 'City',
    woeid: 2487956,
    latt_long: '37.777119, -122.41964'
  }
]

export const weatherPayload = {
  consolidated_weather: [{
    id: 5863259740569600,
    weather_state_name: 'Heavy Cloud',
    weather_state_abbr: 'hc',
    wind_direction_compass: 'W',
    created: '2020-12-06T06:34:14.572087Z',
    applicable_date: '2020-12-06',
    min_temp: 5.41,
    max_temp: 16.305,
    the_temp: 15.059999999999999,
    wind_speed: 4.547368277541823,
    wind_direction: 276,
    air_pressure: 1024.5,
    humidity: 49,
    visibility: 17.01345392905432,
    predictability: 71,
  }]
}
