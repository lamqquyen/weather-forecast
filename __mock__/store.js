import thunk from 'redux-thunk'
import configureMockStore from 'redux-mock-store' 

const mockStore = configureMockStore([thunk])
const initialState = {
  location: {
    queryResults: [],
    loading: false
  },
  weather: {
    info: {}
  }
}

export {mockStore, initialState}
