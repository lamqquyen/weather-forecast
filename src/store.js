import {createStore, applyMiddleware} from 'redux'
import thunk from 'redux-thunk'
import allReducers from './reducers'
import {composeWithDevTools} from 'redux-devtools-extension'

const composeEnhancer = composeWithDevTools(applyMiddleware(thunk))
const store = createStore(allReducers, composeEnhancer);

export default store
