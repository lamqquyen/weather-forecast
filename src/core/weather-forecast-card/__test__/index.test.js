import React from 'react'
import WeatherForecastCard from '../index'
import {Provider} from 'react-redux'
import TestRenderer from 'react-test-renderer';
import {mockStore, initialState} from '../../../../__mock__/store'
import SearchField from "../../../components/search-field";

describe("WeatherForecastCard", () => {
  const store = mockStore(initialState)

  it('renders correctly', () => {

    const tree = TestRenderer.create(
      <Provider store={store}>
        <WeatherForecastCard />
      </Provider>
    ).toJSON()

    expect(tree).toMatchSnapshot();
  })

  it('type value', () => {
    const tree = TestRenderer.create(
      <Provider store={store}>
        <WeatherForecastCard />
      </Provider>
    ).root

    const {act} = TestRenderer
    const searchInput = tree.findByType(SearchField)

    act(() => {
      const e = {target: {value: 'test'}}
      searchInput.props.onChange(e)
    })

    expect(searchInput.props.value).toEqual('test')
  })
})