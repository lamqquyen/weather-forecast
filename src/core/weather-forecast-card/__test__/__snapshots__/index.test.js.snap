// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`WeatherForecastCard renders correctly 1`] = `
<div
  className="weather-forecast__wrapper"
>
  <h2
    className="weather-forecast__title"
  >
    WEATHER FORECAST
  </h2>
  <div
    className="row"
  >
    <div
      className="col-md-4 offset-md-1"
    >
      <div
        className="weather-forecast__input input-group"
      >
        <div
          className="input-group-prepend"
        >
          <span
            className="input-group-text"
            id="basic-addon1"
          >
            <i
              className="fa fa-search"
            />
          </span>
        </div>
        <input
          className="form-control"
          onChange={[Function]}
          onFocus={[Function]}
          placeholder="Search"
          value=""
        />
      </div>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-md-10 offset-md-1"
    >
      <div
        className="weather-block__wrapper"
      >
        <div>
          Search a location to get weather informations
        </div>
      </div>
    </div>
  </div>
</div>
`;
